homeassistant/components/vesync/fan.py

```diff
@@ -0,0 +1,114 @@
+"""Support for VeSync fans."""
+import logging
+
+from homeassistant.components.fan import (
+    SPEED_HIGH,
+    SPEED_LOW,
+    SPEED_MEDIUM,
+    SPEED_OFF,
+    SUPPORT_SET_SPEED,
+    FanEntity,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .common import VeSyncDevice
+from .const import DOMAIN, VS_DISCOVERY, VS_DISPATCHERS, VS_FANS
+
+_LOGGER = logging.getLogger(__name__)
+
+DEV_TYPE_TO_HA = {
+    "LV-PUR131S": "fan",
+}
+
+SPEED_AUTO = "auto"
+FAN_SPEEDS = [SPEED_AUTO, SPEED_OFF, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
```

Comment:
`SPEED_AUTO` is not an allowed value in the `speed_list`
https://developers.home-assistant.io/docs/core/entity/fan/#properties

This is currently an architecture discussion on how to handle `auto` / `smart` speeds here 
https://github.com/home-assistant/architecture/issues/127

As the codebase currently stands, adding `auto` causes problems for `homekit` and other integrations, so please take it out.

Reply:
Removed for now. I read through that discussion link and will follow for updates to know when / how to account for auto.